<launch>
<arg name="name" />
<arg name="device" />
<arg name="topic_name" />
<arg name="topic" />

<group ns="$(arg name)" >
  <node pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera" output="screen">
    <param name="width" type="int" value="1280" /> 
    <param name="height" type="int" value="720" />
    <param name="fps" type="int" value="8" />
    <param name="frame" type="string" value="wide_stereo" />
    <param name="device" type="string" value="$(arg device)" /> 

    <remap from="image_raw" to="image_raw" />
    <remap from="camera_info" to="camera_info" />

  </node>

  <node ns="camera" pkg="image_proc" type="image_proc" name="image_proc" if="0"/>
    <group>
      <!-- node to reduce the size of raw camera image -->
      <node pkg="nodelet" type="nodelet"
        args="standalone image_proc/crop_decimate"
        name="decimator">

        <param name="decimation_x" type="int" value="3" />
        <param name="decimation_y" type="int" value="3" />

        <remap from="camera/image_raw" to="image_raw" />
      </node>

      <node pkg="nodelet" type="nodelet"
        args="standalone image_proc/debayer"
        name="debayer">
        <remap from="image_raw" to="camera_out/image_raw"/>
      </node>


      <node pkg="rover_camera" type="reduce_fps.py" name="reduce_fps">
        <param name="_fps" type="int" value="10" />
        <remap from="/image_mono/compressed" to="/$(arg name)/image_mono/compressed" />
        
        <remap if="$(arg topic)" from="/reduced/compressed" to="/$(arg topic_name)" />
      </node>

    </group>
</group>

</launch>
